noodles-bed/src/de.rs

```diff
@@ -0,0 +1,934 @@
+use std::fmt;
+use std::io::BufRead;
+use std::ops::{AddAssign, MulAssign, Neg};
+
+use noodles_core::Position;
+
+use serde::Deserialize;
+use serde::de::{
+    self, DeserializeSeed, EnumAccess, IntoDeserializer, MapAccess, SeqAccess,
+    VariantAccess, Visitor,
+};
+
+use crate::{Record, reader::Reader, error::{Error, Result}};
+
+const FIELDS: &'static [&'static str] = &["chrom", "start", "end"];
+enum Field { Chrom, Start, End }
```

Comment:
One approach to allow lowercase deserialisation is to use the [rename_all](https://serde.rs/container-attrs.html#rename_all) container attribute. This works for serialisation too.
```rust
#[derive(Deserialize, Debug)]
#[serde(rename_all = "lowercase")]
enum Field { Chrom, Start, End }
```
Note, fields can be renamed arbitrarily using the [rename](https://serde.rs/field-attrs.html#rename) field attribute, or multiple valid names can be assigned using [alias](https://serde.rs/field-attrs.html#alias).